compiler/rustc_hir_analysis/src/collect/lifetimes.rs

```diff
@@ -1834,12 +1834,44 @@ fn is_late_bound_map(tcx: TyCtxt<'_>, def_id: LocalDefId) -> Option<&FxIndexSet<
     debug!(?late_bound);
     return Some(tcx.arena.alloc(late_bound));
 
-    #[derive(Default)]
-    struct ConstrainedCollector {
+    struct ConstrainedCollectorPostAstConv {
```

Comment:
This needs a lengthy comment linking to the issue and explaining why it exists

Reply:
And explain why we have to use type-of:

* cross-crate (first)
* would need same basic logic for managing cycles (second)